numpy/f2py/cfuncs.py

```diff
@@ -776,8 +776,8 @@ def errmess(s: str) -> None:
         /* empty bytes has trailing null, so dereferencing is always safe */
         *v = PyBytes_AS_STRING(obj)[0];
         return 1;
-    } else if (PyUnicode_Check(obj)) {
-        PyObject* tmp = PyUnicode_AsASCIIString(obj);
```

Comment:
This code still looks quite silly to me; you're using `PyBytes_AS_STRING` below and there is a duplicate check for `PyBytes_Check` above.

Reply:
Can you explain the logic you are trying to implement here?